clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java

```diff
@@ -424,7 +446,8 @@ private void resetOffsetPosition(TopicPartition tp) {
             if (offset == null)
                 throw new IllegalStateException("MockConsumer didn't have beginning offset specified, but tried to seek to beginning");
         } else if (strategy == OffsetResetStrategy.LATEST) {
-            offset = endOffsets.get(tp);
+            List<Long> offsets = endOffsets.get(tp);
+            offset = offsets.size() > 1 ? offsets.remove(0) : offsets.get(0);
```

Comment:
as above. don't we nee a check for `offset == null` here, too?

Reply:
ack